libraries/AP_HAL_ChibiOS/CANFDIface.cpp

```diff
@@ -378,7 +378,8 @@ int16_t CANIface::send(const AP_HAL::CANFrame& frame, uint64_t tx_deadline,
     pending_tx_[index].aborted        = false;
     pending_tx_[index].setup          = true;
     pending_tx_[index].pushed         = false;
-    return 1;
+
+    return AP_HAL::CANIface::send(frame, tx_deadline, flags);
```

Comment:
i'm not sure about this. It would mean when using gui_tool as a bus analyser we could get a misleading impression of what is being sent on the bus, as we may see packets that are not actually being sent
for now this does seem to work well with nothing on the bus. The only other thing I can think of is to make the behaviour depend on the target node.